# Sentinel-2 Download via Copernicus

Sentinel-2 satelliete data was collected via [Copernicus Data Space Ecosystem](https://dataspace.copernicus.eu/data-collections/copernicus-sentinel-missions/sentinel-2). Downloaded for the years 2015-2024, around the AOI: Upper Keys Florida particularly around John Pennekamp Coral Reef State Park.

1. Copernicus
2. Create AOI - using coordinates of bounding box in JSON
3. `Search` Tab Sentinel-2 Data
    - Type: 
        L1C for 2015-2016
        L1A for 2017-2024
        - Filters: **L2A is checked**

Note: for the fastest loading and most effecient way to add the scenes to workspace I found was to search years in groups of two.



### Imports

In [1]:
import geopandas as gpd
from shapely.geometry import box

## Coordinates of AOI

In [2]:
# Load shapefile
gdf = gpd.read_file('C:/Users/sulli/DATA_Science/seagrass_project/study_area/UpperKeys_FL.shp')

# Ensure correct CRS
if gdf.crs != "EPSG:4326":
    gdf = gdf.to_crs("EPSG:4326")

# Get bounding box
minx, miny, maxx, maxy = gdf.total_bounds

# Create bbox polygon (optional)
bbox_poly = box(minx, miny, maxx, maxy)
bbox_gdf = gpd.GeoDataFrame(geometry=[bbox_poly], crs="EPSG:4326")

print(minx, miny, maxx, maxy)

-80.622018899028 24.849346948181537 -80.13600456261818 25.374374778144837


### JSON to create AOI in Copernicus

In [12]:
# Reduce number of vertices (tolerance in degrees, e.g., 0.001 ~ 100 m)
gdf_simplified = gdf.copy()
gdf_simplified['geometry'] = gdf_simplified.geometry.simplify(tolerance=0.001)

# Convert to GeoJSON
aoi_geometry_simplified = gdf_simplified.geometry.iloc[0].__geo_interface__

import json
aoi_json_simplified = json.dumps(aoi_geometry_simplified, indent=2)
print(aoi_json_simplified)

{
  "type": "Polygon",
  "coordinates": [
    [
      [
        -80.24325373203536,
        25.335168200356744
      ],
      [
        -80.20947649020067,
        25.335984142861875
      ],
      [
        -80.19904087242743,
        25.323793526269164
      ],
      [
        -80.13600456261818,
        25.261835266470268
      ],
      [
        -80.2164381466721,
        25.11706877677012
      ],
      [
        -80.34977305459458,
        24.950408986529432
      ],
      [
        -80.52383791183986,
        24.849346948181537
      ],
      [
        -80.59515512214156,
        24.9303616391827
      ],
      [
        -80.6131870741021,
        24.941505924999372
      ],
      [
        -80.6143470702525,
        24.952615810404996
      ],
      [
        -80.622018899028,
        24.971073394521067
      ],
      [
        -80.56479523543781,
        24.996229689602437
      ],
      [
        -80.56085800841491,
        24.99904931802235
      ],
      [
        -80.55782

#### Resulting JSON Coordinates:
<div style="text-align: center;">
    <code>"spatial_extent": {
  "type": "Polygon",
  "coordinates": [
    [
      [-80.622, 24.849],
      [-80.622, 25.374],
      [-80.136, 25.374],
      [-80.136, 24.849],
      [-80.622, 24.849]
    ]
  ]
}</code>
</div>
